include/NetworkDevice.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "Packet.hpp"
+
+#include <vector>
+#include <memory>
+#include <cstdint>
+#include <unordered_map>
+
+enum class DeviceType: std::uint8_t {
+    Switch,
+    ComputerSenderNode,
+    ComputerReceiverNode,
+};
+
+class NetworkDevice {
+public:
+    explicit NetworkDevice(DeviceType type, std::uint32_t id);
+    virtual ~NetworkDevice() = default;
+
+    /**
+     * @brief Vector of pointers to adjacent network devices.
+     *
+     * @details Usage of std::weak_ptr is necessary to avoid cyclic links.
+     */
+    std::vector<std::weak_ptr<NetworkDevice>> adjacent_devices;
+
+    /**
+     * @brief Network device type.
+     */
+    DeviceType type;
+
+    /**
+     * @brief Network device ID.
+     */
+    std::uint32_t id;
+
+    /**
+     * @brief Network device storage.
+     *
+     * @details Device stores packets from adjacent device in corresponding vector. \n
+     * Key: adjacent device ID. Value: vector with packets.
+     */
+    std::unordered_map<std::uint32_t, std::vector<Packet>> storage;
```

Comment:
`storage` here need clarification. What is the design idea that you implement here? What do you thinks about introducing ingress/egress queues instead?

Reply:
I think that idea of ingress/egress queues is better than `storage` idea.  `storage` -- is an ingress container, I thought it was a good way to store incoming packets, but now I realize that it will be more slower than priority queue.

We can use `priority_queue<Packet>` as ingress container and `queue<Packet>` as egress container.

Reply:
> We can use priority_queue<Packet> 

As discussed elsewhere (see issue #12), `priority_queue` should be in the global event scheduler.